cmd/pdatagen/internal/base_fields.go

```diff
@@ -462,3 +474,77 @@ func (pf *primitiveAsDoubleField) generateCopyToValue(sb *strings.Builder) {
 }
 
 var _ baseField = (*primitiveField)(nil)
+
+type field struct {
+	name            string
+	defaultVal      string
+	testVal         string
+	returnType      string
+	originFieldName string
+	originFullName  string
+	asName          string
+}
+type numberField struct {
+	fields []field
+}
+
+func (pf *numberField) generateAccessors(ms baseStruct, sb *strings.Builder) {
+	for _, field := range pf.fields {
+		sb.WriteString(os.Expand(accessorsNumberTemplate, func(name string) string {
+			switch name {
+			case "structName":
+				return ms.getName()
+			case "fieldName":
+				return field.name
+			case "lowerFieldName":
+				return strings.ToLower(field.name)
+			case "returnType":
+				return field.returnType
+			case "originFieldName":
+				return field.originFieldName
+			case "originFullName":
+				return field.originFullName
+			case "asName":
+				return field.asName
+			default:
+				panic(name)
+			}
+		}))
+		sb.WriteString("\n")
+	}
+}
```

Comment:
I would split the current `generateAccessors` into 2:

```golang
func (opv *oneofPrimitiveValue) generateAccessors(ms baseStruct, sb *strings.Builder) {
	sb.WriteString(os.Expand(accessorsNumberTemplate, func(name string) string {
		switch name {
		case "structName":
		return ms.getName()
		case "fieldName":
			return field.name
		case "lowerFieldName":
			return strings.ToLower(field.name)
		case "returnType":
			return field.returnType
		case "originFieldName":
			return field.originFieldName
		case "originFullName":
			return field.originFullName
		case "asName":
			return field.asName
		default:
			panic(name)
		}
	}))
	sb.WriteString("\n")
}

func (nf *numberField) generateAccessors(ms baseStruct, sb *strings.Builder) {
	for _, field := range nf.fields {
		field.generateAccessors(ms, sb)
	}
}
```

Reply:
Done, I did the same for `generateAccessorsTest`